tap_stripe/__init__.py

```diff
@@ -468,6 +469,23 @@ def convert_dict_to_stripe_object(record):
 
     return record
 
+
+def invoice_line_item_not_found_error(error):
+    """
+        This function checks whether the error contains 'does not have a line item' substring and
+        return boolean values accordingly, to decide whether to backoff or not.
+    """
+    # retry if the error string contains 'does not have a line item'
+    if str(error).__contains__('does not have a line item'):
+        return False
+    return True
+
+
+@backoff.on_exception(backoff.expo,
+                        stripe.error.InvalidRequestError,
+                        giveup=invoice_line_item_not_found_error,
+                        max_tries=5,
+                        factor=2)
```

Comment:
@prijendev Is there any reason to put backoff on whole stream sync again rather than any API call or SDK request?

Reply:
SDK return error when we iterate over response. So, back off on a particular method or SDK request will not work.